pkg/cmd/adm/register_member.go

```diff
@@ -120,59 +119,224 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToDetails.KubeClient, kubeConfigSecret, func() error {
+
+		// update the secret label
+		labels := kubeConfigSecret.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[toolchainv1alpha1.ToolchainClusterLabel] = joiningClusterDetails.Name
+		kubeConfigSecret.Labels = labels
+
+		// update the kubeconfig data
+		kubeConfigSecret.StringData = map[string]string{
+			"kubeconfig": string(generatedKubeConfigFormatted),
+		}
+
+		return nil
+	})
+
 	if err != nil {
-		return nil, errs.Wrapf(err, "unable to get add-script.sh")
-	}
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return nil, fmt.Errorf("unable to get add-script.sh - response status %s", resp.Status)
+		return err
 	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil {
-			term.Printlnf(err.Error())
+	term.Println("Secret successfully reconciled")
+	term.Printlnf("operation", op)
+
+	// TODO -- temporary logic
+	// The creation of the toolchaincluster is just temporary until we implement https://issues.redhat.com/browse/KUBESAW-44,
+	// the creation logic will be moved to the toolchaincluster_resource controller in toolchain-common and will be based on the secret created above.
+	//
+	// create/update toolchaincluster on the clusterJoinTo
+	term.Printlnf("creating ToolchainCluster representation of %s in %s:", joiningClusterType, clusterJoinToDetails.Name)
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: joiningClusterDetails.Name, Namespace: clusterJoinToDetails.OperatorNamespace}}
+	op, err = controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToDetails.KubeClient, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels["namespace"] = joiningClusterDetails.OperatorNamespace
+		if joiningClusterType == "member" {
+			labels["cluster-role.toolchain.dev.openshift.com/tenant"] = ""
+		}
+
+		toolchainClusterCR.Labels = labels
+		toolchainClusterCR.Spec.APIEndpoint = joiningClusterDetails.APIEndpoint
+		toolchainClusterCR.Spec.SecretRef.Name = secretName
+		if insecureSkipTLSVerify {
+			toolchainClusterCR.Spec.DisabledTLSValidations = []toolchainv1alpha1.TLSValidation{toolchainv1alpha1.TLSAll}
 		}
-	}()
-	// Create the file
-	file, err := os.CreateTemp("", "add-cluster-*.sh")
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+	term.Println("Toolchaincluster successfully reconciled")
+	term.Printlnf("operation", op)
+	// -- end temporary logic
+
+	return err
+}
+
+func secretName(SANamespacedName runtimeclient.ObjectKey, joiningOperatorNamespace string, joiningClusterName string) string {
+	secretName := SANamespacedName.Name + "-" + joiningOperatorNamespace + "-" + joiningClusterName
```

Comment:
there is something weird with this name - this doesn't match what we have in prod nor in stage. We don't use the namespace in the name there.
Could you please take a closer look why we need this?

Reply:
doesn't seem to be needed anymore  see https://redhat-internal.slack.com/archives/C06MJ2DVBU4/p1723472161980079

dropped in https://github.com/kubesaw/ksctl/pull/53/commits/de3e9f8dcac29c7d63f67f64f2084f43e8134bf3